src/shared/sam/localLambdaRunner.ts

```diff
@@ -45,11 +45,20 @@ export interface SAMTemplateEnvironmentVariables {
     }
 }
 
-function getEnvironmentVariables(env?: { [k: string]: string | number | boolean }): SAMTemplateEnvironmentVariables {
-    return env ? { [TEMPLATE_RESOURCE_NAME]: env } : {}
+function getEnvironmentVariables(
+    resourceName: string,
+    env?: { [k: string]: string | number | boolean }
+): SAMTemplateEnvironmentVariables {
+    return env ? { [resourceName]: env } : {}
+}
+
+/**
+ * Decides the resource name for the generated template.yaml.
+ */
+function makeResourceName(config: SamLaunchRequestArgs): string {
+    return config.invokeTarget.target === 'code' ? 'awsToolkitSamLocalResource' : config.invokeTarget.logicalId
```

Comment:
Consider using `TEMPLATE_RESOURCE_NAME`

Reply:
I deleted `TEMPLATE_RESOURCE_NAME` because it isn't used anywhere, `makeResourceName()` serves the same purpose.